Spanner/src/SpannerClient.php

```diff
@@ -325,6 +376,44 @@ public function instance($name, array $instance = [])
             $this->returnInt64AsObject,
             $instance
         );
+
+        if (!isset($this->config['apiEndpoint']) &&
+            getenv('GOOGLE_CLOUD_ENABLE_RESOURCE_BASED_ROUTING')
+        ) {
+            if (!isset($this->endpoints[$instanceObject->name()])) {
+                $instanceInfo = $instanceObject->info([
+                    'fieldMask' => new FieldMask(['paths' => ['endpoint_uris']])
+                ]);
+                $this->endpoints[$instanceObject->name()] = $instanceInfo['endpointUris'];
+            }
+
+            if (!is_null($apiEndpoint = $this->endpoint($instanceObject->name()))) {
+                $connection = $this->createConnection(['apiEndpoint' => $apiEndpoint]);
+                list($lroConnection, $lroCallables) = $this->createLroProperties($connection);
+                $instanceObject->connectionUpdate($connection, $lroConnection, $lroCallables);
```

Comment:
I assume, old gRPC connection should be closed somehow, not just replaced with another connection object

Reply:
I think the old gRPC connection can still be used to request the next list of endpoints while creating the next instance. Therefore, it is not redefined for the SpannerClient object. In addition, in the Grpc class and traits, the close method is not available.

Reply:
Yes, but you're doing:
```
$this->connection = $new_connection
```
Where `$this->connection' is `SpannerClient()` connection passed into instance in here:
https://github.com/q-logic/google-cloud-php/blob/1ed49dd7dde7dbcde018c5f29f5b3e8f86dc2f15/Spanner/src/SpannerClient.php#L319-L327
All objects are getting connection this way, so, I suppose, overriding it in one place overrides connection in all objects at once